src/control/tools/StrokeHandler.cpp

```diff
@@ -334,7 +308,7 @@ void StrokeHandler::onButtonPressEvent(const PositionInputData& pos) {
     double width = page->getWidth() * zoom * dpiScaleFactor;
     double height = page->getHeight() * zoom * dpiScaleFactor;
 
-    surfMask = cairo_image_surface_create(CAIRO_FORMAT_A8, width, height);
+    surfMask = cairo_image_surface_create(CAIRO_FORMAT_A8, (int)std::ceil(width), (int)std::ceil(height));
```

Comment:
Made this change too. It could be relevant if Cairo reacts badly when try to paint outside the bounds of the surface. I don't know.